### PR TITLE
Improve ChangeTracker expiry

### PR DESCRIPTION
**Pull request by almirsarajcic**

### CODE REVIEW

File: lib/chat/db/change_tracker/tracking.ex [hunk 1]
```diff
@@ -120,4 +114,16 @@ defmodule Chat.Db.ChangeTracker.Tracking do
     state
     |> tracker(next_id: id + 1, keys: new_keys, items: items |> Map.put(id, item))
   end
+
+  defp expiry_callback(changes, :dev) when length(changes) > 0 do
```
Comment (sergey-lukianov) [inline lib/chat/db/change_tracker/tracking.ex:118]: I do not want this checks to be done in runtime. We can place them in compile time

Also `changes != []` is faster than `length(change)` I guess

**Review by sergey-lukianov — CHANGES_REQUESTED** [review]

File: lib/chat/db/change_tracker/tracking.ex (hunk 1 above)
Comment (almirsarajcic) [reply lib/chat/db/change_tracker/tracking.ex:118]: I understand. Dialyzer will complain, though, that the dev/test part of the code can never get executed.
But I'll change it for now. When we setup Dialyzer in CI we'll need to figure that out.

Comment (sergey-lukianov) [reply lib/chat/db/change_tracker/tracking.ex]: now it is compiled in something like
```
...
defp expiry_callback do
 case :dev do
   :test -> refute
   :dev -> error
   _ -> nil
 end
end
```

The idea is to create a macro that will change function body depending on environment it  is compiled for

**Review by sergey-lukianov — CHANGES_REQUESTED** [review]
A macro can do the job

File: lib/chat/db/change_tracker/tracking.ex (hunk 1 above)
Comment (almirsarajcic) [reply lib/chat/db/change_tracker/tracking.ex]: Ah, now I get it 😃

**Review by sergey-lukianov — APPROVED** [review]

**Pull request merged**